src/source/Metrics/Metrics.c

```diff
@@ -7,11 +7,48 @@
 STATUS getIceCandidatePairStats(PRtcPeerConnection pRtcPeerConnection, PRtcIceCandidatePairStats pRtcIceCandidatePairStats)
 {
     STATUS retStatus = STATUS_SUCCESS;
-    PKvsPeerConnection pKvsPeerConnection = (PKvsPeerConnection) pRtcPeerConnection;
-    UNUSED_PARAM(pKvsPeerConnection);
+    BOOL locked = FALSE;
+    PIceAgent pIceAgent = ((PKvsPeerConnection) pRtcPeerConnection)->pIceAgent;
     CHK((pRtcPeerConnection != NULL || pRtcIceCandidatePairStats != NULL), STATUS_NULL_ARG);
-    CHK(FALSE, STATUS_NOT_IMPLEMENTED);
+    MUTEX_LOCK(pIceAgent->lock);
+    locked = TRUE;
+    if (pIceAgent->pDataSendingIceCandidatePair == NULL) {
+        CHK(FALSE, STATUS_SUCCESS);
+    }
```

Comment:
instead, use CHK(pIceAgent->pDataSendingIceCandidatePair != NULL, STATUS_SUCCESS) if you want an early success return

Reply:
Done.

Reply:
Done